src/instructor.ts

```diff
@@ -246,16 +251,20 @@ class Instructor {
     response_model,
     ...params
   }: ChatCompletionCreateParamsWithModel<T>): ChatCompletionCreateParams => {
-    const jsonSchema = zodToJsonSchema(response_model.schema, {
-      name: "response_model",
+    // TODO: update default response_model name
+    const model_name = response_model.name ?? "response_model"
+
+    const modelJsonSchema = zodToJsonSchema(response_model.schema, {
+      name: model_name,
       errorMessages: true
-    })
+    }).definitions
 
-    this.log("JSON Schema from zod: ", jsonSchema)
+    this.log("JSON Schema from zod: ", modelJsonSchema)
 
     const definition = {
-      name: "response_model",
-      ...jsonSchema.definitions?.response_model
+      name: model_name,
+      description: response_model.description,
```

Comment:
AH, I see now - we were passing the top level schema vs just the definitions - makes sense now - super curious though now as to why they still work - 

 I'd say the only thing would be we should prob jsut make sure we have definitions for sure, or maybe just throw explicitly if not here

something like:

```
    const { schema, name = "response_model", description } = response_model

    const { definitions } = zodToJsonSchema(schema, {
      name,
      errorMessages: true
    })

    if (!definitions || !definitions?.[name]) {
      console.warn("Could not extract json schema definitions from your schema", schema)
      throw new Error("Could not extract json schema definitions from your schema")
    }

    this.log("JSON Schema from zod: ", definitions)

    const definition = {
      name,
      description,
      ...definitions[name]
    }

    const paramsForMode = MODE_TO_PARAMS[this.mode](definition, params, this.mode)

    return {
      stream: false,
      ...paramsForMode
    }
    ```

Reply:
oh or I guess no, I read that wrong - I guess I'm just missing where the typo was